src/hooks/useMediaMatch.ts

```diff
@@ -25,9 +25,16 @@ function useMediaMatch(query: string): boolean {
     setMatches(matchMedia.matches);
     const listener = (event_: MediaQueryListEventMap['change']) =>
       setMatches(event_.matches);
-    matchMedia.addEventListener('change', listener);
 
-    return () => matchMedia.removeEventListener('change', listener);
+    if (matchMedia.addEventListener) {
+      matchMedia.addEventListener('change', listener)
```

Comment:
Thanks for this! @krijoh92 

I just have one small suggestion just for consistency in code style that we already have. What do you think of rewriting it to look like this?


```jsx

matchMedia.addEventListener?.('change', listener)
matchMedia.addListener?.(listener)

return () => {
  matchMedia.removeEventListener?.('change', listener) 
  matchMedia.removeListener?.(listener)
}
```

Reply:
I haven't tested yet, but wouldn't this make it trigger twice for all browsers that support `addEventListener` and `addListener`. `addListener` is only deprecated, but not removed in modern browsers

Reply:
Ah. I see. Agreed.